src/config/updateTicks.js

```diff
@@ -6,17 +6,29 @@ import getLastTick from 'binary-utils/lib/getLastTick';
 
 export default (chart, prevProps, nextProps) => {
     const chartType = chart.series[0].type;
+    const { dataMax, min, max } = chart.xAxis[0].getExtremes();
+    let newDataMax = dataMax;
     switch (chartType) {
         case 'area': {
             const oneTickDiff = doTicksDifferJustOneEntry(prevProps.ticks, nextProps.ticks);
             if (oneTickDiff) {
                 const lastTick = getLastTick(nextProps.ticks);
                 const dataPoint = tickToData(lastTick);
+                newDataMax = dataPoint[0];
                 chart.series[0].addPoint(dataPoint, false);
             } else {
                 const dataList = nextProps.ticks.map(tickToData);
+                newDataMax = dataList[dataList.length - 1][0];
                 chart.series[0].setData(dataList, false);
             }
+
+            const frameSize = max - min;
+            const isCloseToMostRecent = (dataMax - max) <= 2000;
```

Comment:
2000 means 2 seconds, which might be too big or too small depends on condition, but we can change when needed